packages/go_router/lib/src/parser.dart

```diff
@@ -124,3 +128,13 @@ class GoRouteInformationParser extends RouteInformationParser<RouteMatchList> {
     );
   }
 }
+
+/// Pre-parsed [RouteInformation] that contains a [RouteMatchList].
```

Comment:
The doc should also mention the how parser will treat the class instances if customer uses this